labstep/__init__.py

```diff
@@ -1,581 +1,646 @@
+
+    Parameters
+    ----------
+    user (obj)
+    The Labstep user adding a tag. Must have property 'api_key'. See 'login'.
+    entity (obj)
+    The Labstep entity to tag. Can be Resource, Experiment, or Protocol. Must have 'id'.
+    tag (str)
     The tag to attach. Must have an 'id' property.
 
-  Returns
-  -------
-  entity
-    Returns the tagged entity.
-  '''
-  if 'experiments' in entity:
-    entityType = 'experiment-workflow'
-  elif 'parent_protocol' in entity:
-    entityType = 'protocol-collection'
-  elif 'resource_location' in entity:
-    entityType = 'resource'
-  elif 'collection' in entity:
-    entityType = 'protocol-collection'
-    entity = entity['collection']
-  else:
-    raise Exception('Entities of this type cannot be tagged')
-
-  url = "https://api.labstep.com/api/generic/"+entityType+"/"+str(entity['id'])+"/tag/"+str(tag['id'])
-  headers = {'apikey': user['api_key']}
-  r = requests.put(url, headers=headers)
-  return json.loads(r.content)
-
-
-
-def getTags(user,name=None,count=1000):
-  '''Retrieve a list of user tags. 
-  
-  Parameters
-  ----------
-  user : obj
-    The labstep user to comment as. Must have property 'api_key'. See 'login'.
-  name : str
-    Search for tags with a specific name.
-  count : int
-    Total number of results to return. Defaults to 1000.
-
-  Returns
-  -------
-  tags
-    A list of tags matching the search query.
-  '''
-  headers = {'apikey': user['api_key']}
-  n = min(count,1000)
-  url = "https://api.labstep.com/api/generic/tag"
-  params = {'search': 1,
-            'cursor':-1,
-            'count':n,
-            }
-
-  if name is not None:
-    params['search_query']=name
-
-  r = requests.get(url, params=params, headers=headers)
-  resp = json.loads(r.content)
-  items = resp['items']
-  expected_results = min(resp['total'],count)
-
-  while len(items)<expected_results:
-    cursor = resp['next_cursor']
-    params= {'search':1,
-             'cursor':cursor,
-             'count':n,
-             }
-    r = requests.get(url, params=params, headers=headers)
-    resp = json.loads(r.content)
-    items.extend(resp['items'])
-    
-  return items
-
-
+    Returns
+    -------
+    entity
+        Returns the tagged entity.
+    '''
+    if 'experiments' in entity:
+        entityType = 'experiment-workflow'
+    elif 'parent_protocol' in entity:
+        entityType = 'protocol-collection'
+    elif 'resource_location' in entity:
+        entityType = 'resource'
+    elif 'collection' in entity:
+        entityType = 'protocol-collection'
+        entity = entity['collection']
+    else:
+        raise Exception('Entities of this type cannot be tagged')
+
+    url = "https://api.labstep.com/api/generic/"+entityType+"/"+str(entity['id'])+"/tag/"+str(tag['id'])
+    headers = {'apikey': user['api_key']}
+    r = requests.put(url, headers=headers)
+    return json.loads(r.content)
 
 def tag(user,entity,name):
```

Comment:
Let's put this function at the bottom. Any functions that don't have a url in them should go at the very bottom. 

At some point we need to split up this file to make it more manageable (something to look into maybe but not super high priority).